### PR TITLE
Martin/ios 231 allow user to delete conversations with long press

### PR DESCRIPTION
**Pull request by MartinJibber**

### CODE REVIEW

File: iOS/Flows/Conversation/Cells/ConversationMessageCell.swift [hunk 1]
```diff
@@ -276,19 +276,19 @@ extension ConversationMessageCell: UIContextMenuInteractionDelegate {
             self.handleDeleteMessage?(message)
         }
 
-        let deleteMenu = UIMenu(title: "Delete Thread",
+
+        let deleteText = message.isFromCurrentUser ? "Delete Conversation" : "Hide Conversation"
+        let deleteMenu = UIMenu(title: deleteText,
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/ConversationMessageCell.swift]: Making the delete text contextual based on the user's permissions.

File: iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift [hunk 2]
```diff
@@ -18,10 +18,27 @@ extension ConversationListViewController {
...
+        self.dataSource.handleDeleteMessage = { (conversation) in
+            Task {
+                do {
+                    let cid = try ConversationID(cid: conversation.conversationId)
+                    let conversationController = ChatClient.shared.channelController(for: cid)
+                    if conversationController.conversation.isFromCurrentUser {
+                        try await conversationController.deleteChannel()
+                    } else {
+                        try await conversationController.hideChannel()
+                    }
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift]: Delete or hide the conversation based on permissions. Hiding means the conversation won't be shown until someone else sends a message to it.

File: iOS/_Stream Temp/ConversationController+Extensions.swift [hunk 3]
```diff
@@ -355,6 +355,51 @@ extension ChatChannelController {
         }
     }
 
+    /// Remove users to the channel as members.
```
Comment (MartinJibber) [inline iOS/_Stream Temp/ConversationController+Extensions.swift:372]: Async extensions.

File: iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift (hunk 2 above)
Comment (Benjamindodgson) [reply iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift]: Does this need be added to a task pool?

**Review by Benjamindodgson — APPROVED** [review]
1 change

**Pull request merged**